Userland/Libraries/LibJS/AST.cpp

```diff
@@ -158,6 +158,11 @@ CallExpression::ThisAndCallee CallExpression::compute_this_and_callee(Interprete
 
         return { this_value, callee };
     }
+
+    if (interpreter.vm().in_strict_mode())
+        // If we are in strict mode, |this| should never be bound to global object by default.
+        return { js_undefined(), m_callee->execute(interpreter, global_object) };
```

Comment:
Please use curly braces for any if/else branches where the body is more than one line:

```suggestion
    if (interpreter.vm().in_strict_mode()) {
        // If we are in strict mode, |this| should never be bound to global object by default.
        return { js_undefined(), m_callee->execute(interpreter, global_object) };
    }
```